### ISSUE TITLE
Difference between SourceTree and tool statistic

### ISSUE THREAD

**Issue by max-mykhailenko** [ejwa/gitinspector#94]
Some people in tool statistic have only 2 commits, but SourceTree in search by user give more than 50 commits. How I can debug this moment?

**Comment by adam-waldenberg** [comment]
Hi @max-mykhailenko.

Most likely, gitinspector is doing exactly what it was made for. This question has come up several times in the past. See the last question covered in the [FAQ](https://github.com/ejwa/gitinspector/wiki/FAQ). Also see issues #75 and #84.

**Comment by adam-waldenberg** [comment]
@max-mykhailenko

A tip could be to do "-f **" on the command line when starting gitinspector. This will include all commits regardless of what file extensions are covered in them. Most of the time, this is not desirable, though. As it also includes different text files and documents in the analysis.

You could also do "-l" in order to see which file extensions are being detected by gitinspector. This will also show you which ones are included in the current analysis.

**Comment by max-mykhailenko** [comment]
`-l` option give me the next result. Project contains php, js, twig. I'm sure that statistic is wrong for php.

```
The extensions below were found in the repository history (extensions used
during statistical analysis are marked):
* crt css html [js] json less log map markdown md net Premium 2048 Secure Server CA pem php po sh svg twig txt xml yml 
```

`gitinspector -f ** ../holdshort.com/`

give me 

```
Traceback (most recent call last):
  File "/usr/local/bin/gitinspector", line 9, in <module>
    load_entry_point('gitinspector==0.5.0dev', 'console_scripts', 'gitinspector')()
  File "/Library/Python/2.7/site-packages/gitinspector-0.5.0dev-py2.7.egg/gitinspector/gitinspector.py", line 140, in main
    repos = __get_validated_git_repos__(set(args))
  File "/Library/Python/2.7/site-packages/gitinspector-0.5.0dev-py2.7.egg/gitinspector/gitinspector.py", line 119, in __get_validated_git_repos__
    basedir_path = basedir.get_basedir_git(cloned_repo.location)
  File "/Library/Python/2.7/site-packages/gitinspector-0.5.0dev-py2.7.egg/gitinspector/basedir.py", line 35, in get_basedir_git
    os.chdir(path)
OSError: [Errno 20] Not a directory: '/Users/max/Work/Projects/gitinspector/stdeb.cfg'
```

**Comment by max-mykhailenko** [comment]
`gitinspector --file-types=php ../holdshort.com/` give right statistic for that user

**Comment by max-mykhailenko** [comment]
I understand, documentation not so clearly as can be.

need to run `gitinspector --file-types='**' ... `

**Comment by max-mykhailenko** [comment]
No, I don't understand.
I use `--file-types='**'` but see only 14 commits by the user. But in that period user has more than 200 commits.
Extensions included correctly

```
The extensions below were found in the repository history (extensions used during statistical analysis are marked):
[css] [js] [json] [less] [map] [php] [twig] 
```

How this is possible?

**Comment by adam-waldenberg** [comment]
@max-mykhailenko 

Let's see... Using `-f php` gives the correct behavior, while `-f "**"` is missing rows? That sounds unlikely.
Don't forget that  pure binary commits also tend to be skipped.

Support for `-f "**"` and `-f "*"` are fairly new and not thoroughly tested. I actually found a bug with `-f "*"`. It's including some commits with binary files - something it should not be doing.

**Comment by max-mykhailenko** [comment]
@adam-waldenberg I will try to write all extensions to `--file-types` and write results above

**Comment by adam-waldenberg** [comment]
@max-mykhailenko 

Opened issue #95 which directly covers the bug with `-f "**"`.

**Comment by max-mykhailenko** [comment]
`file-types` give the same result as --file-types='**' except but without one commit. I see only 13 commits instead more than 200

`gitinspector -T -l --since=2015/11/01 --file-types=php,js,twig ../holdshort.com/`

<img width="930" alt="screen shot 2015-12-17 at 4 02 47 pm" src="https://cloud.githubusercontent.com/assets/1213198/11871387/f1f5474a-a4d7-11e5-9c4c-9a884fe16650.png">
<img width="916" alt="screen shot 2015-12-17 at 4 04 21 pm" src="https://cloud.githubusercontent.com/assets/1213198/11871388/f2215cfe-a4d7-11e5-90e2-6c27f8465421.png">

**Comment by adam-waldenberg** [comment]
@max-mykhailenko 

What about the css,json,less,map file extensions?
So there should be over 200 commits by Evgeniy Dolgiy since November 1st?

**Comment by max-mykhailenko** [comment]
@adam-waldenberg 
I add that extensions but result is the same
`gitinspector -T -l --since=2015/11/01 --file-types=php,js,twig,css,map,less ../holdshort.com/`

```
Author                     Commits    Insertions      Deletions    % of changes
Evgeniy Dolgiy                  13           133            137           41.86
```

**Comment by adam-waldenberg** [comment]
In that case it seems something is causing gitinspector to filter or skip those commits... The question is what. I have not seen this behavior before. I need a repository behaving similarly in order to debug it. 

You could try and see if you can find what is going on. You should be able to track down what is happening with some `print()` calls inside `gitinspector/changes.py`.

**Comment by max-mykhailenko** [comment]
@adam-waldenberg okey, no problem, thanks!

**Comment by adam-waldenberg** [comment]
First thing to do could be to try the git command gitinspector runs;

```python
git_log_r = subprocess.Popen(filter(None, ["git", "log", "--reverse", "--pretty=%ct|%cd|%H|%aN|%aE",
		"--stat=100000,8192", "--no-merges", "-w", interval.get_since(),
		interval.get_until(), "--date=short"] + (["-C", "-C", "-M"] if self.hard else []) +
		[self.first_hash + self.second_hash]), bufsize=1,    
		stdout=subprocess.PIPE).stdout
```

on the command line in order to check what you get.

**Comment by max-mykhailenko** [comment]
This git log give same 14 commits

```
git log --reverse --pretty --stat=100000,8192 --no-merges -w --after="2015-11-01" --date=short --author="Evgeniy Dolgiy"
```

But if I add ` --all` I see much more commits.

```
--all
Pretend as if all the refs in refs/ are listed on the command line as <commit>.
```

**Comment by max-mykhailenko** [comment]
What this mean from practical point of view I don't know. Want to hear your thought about it.
Thank you for your patience

**Comment by adam-waldenberg** [comment]
@max-mykhailenko 

AFAIK --all also includes commits from branches reachable in refs/ (not only the current one). 
As of now, gitinspector only gathers commits from the currently active branch. See issue #92 for more information on this.

**Comment by max-mykhailenko** [comment]
Now I see. In our team for every task we create branch and get statistic separately for each branch is impossible because we remove branch after issue was solved.

I will read documentation more accurately in the next time.

Thanks.

**Comment by adam-waldenberg** [comment]
@max-mykhailenko 

That would be a problem, yes. I any case, I'm marking this as invalid and closing :).
